### PR TITLE
studio: Add DVC Studio commands to documentation

### LINKED ISSUE #4979: Add documentation for dvc studio command
### Report

Related PR in dvc: https://github.com/iterative/dvc/pull/10074

### PR DESCRIPTION
**Pull request by amritghimire**
> You may disregard these recommendations if you used the **Edit on GitHub** button from dvc.org to improve a doc in place.

❗ Please read the guidelines in the [Contributing to the Documentation](https://dvc.org/doc/user-guide/contributing/docs) list if you make any substantial changes to the documentation or JS engine.

🐛 Please make sure to mention `Fix #issue` (if applicable) in the description of the PR. This causes GitHub to close it automatically when the PR is merged.

Please choose to [allow us](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork) to edit your branch when creating the PR.

Add guides for dvc studio login feature introduced in  dvc: https://github.com/iterative/dvc/pull/10074

Closes #4979

Supersedes #4981

### CODE REVIEW

File: content/docs/command-reference/studio/index.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+## Description
+
+`dvc studio` authorize dvc with Studio and set the token. When this is set, DVC
+uses this to share live experiments and notify Studio about pushed experiments.
+
+## Options
+
+- `-h`, `--help` - show this help message and exit
+- `-q`, `--quiet` - Be quiet
```
Comment (shcheklein) [inline content/docs/command-reference/studio/index.md:36]: style. check if in other places it starts with capital letter or not. If there are spaced between items in the list, etc

File: content/docs/command-reference/studio/login.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+## studio login
+
+Authorize DVC with Studio to save the token to global [DVC configuration].
```
Comment (shcheklein) [inline content/docs/command-reference/studio/login.md:3]: Check terminology.

File: content/docs/command-reference/studio/login.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+## Synopsis
+
+```usage
+usage: dvc studio login [-h] [-q | -v] [-H <hostname>] [-s <scopes>] [-n <name>] [-d]
+```
+
+## Description
+
+By default, this command authorize dvc with Studio with default scopes and a
```
Comment (shcheklein) [inline content/docs/command-reference/studio/login.md:16]: dvc -> DVC

File: content/docs/command-reference/studio/logout.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+## studio logout
+
+This command helps to log out user from DVC Studio.
```
Comment (shcheklein) [inline content/docs/command-reference/studio/logout.md:3]: style: helps -> Log out user from DVC Studio. (simpler is better, no need to use `helps`, etc).

File: content/docs/command-reference/studio/token.md [hunk 5]
```diff
@@ -0,0 +1,22 @@
+## studio token
+
+This command helps to log out user from DVC Studio.
```
Comment (shcheklein) [inline content/docs/command-reference/studio/token.md:3]: Probably wrong description

File: content/docs/command-reference/studio/index.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+```usage
+usage: dvc studio [-h] [-q | -v] {login,logout,token} ...
+
+positional arguments:
+  {login,logout,token}  Use `dvc studio CMD --help` to display command-specific help.
+    login               Authenticate DVC with Studio host
+    logout              Logout user from Studio
+    token               View the token dvc uses to contact Studio
+
+options:
```
Comment (shcheklein) [inline content/docs/command-reference/studio/index.md:22]: Style: we don't include option descriptions in the usage section, probably we also don't include positional arg descriptions

**Review by shcheklein — CHANGES_REQUESTED** [review]
Let's please review terminology (and in the product as well if needed) + please check the style / details - compare with `dvc add` or other commands that we have for  a while.

File: content/docs/command-reference/studio/login.md [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+- `-H <hostname>`, `--hostname <hostname>` - The hostname of the Studio instance
+  to authenticate with.
+- `-s <scopes>`, `--scopes <scopes>` - The scopes for the authentication token.
+- `-n NAME`, `--name <name>` - The name of the authentication token. It will be
+  used to identify token shown in Studio profile.
+- `-d`, `--use-device-code` - Use authentication flow based on user code. You
+  will be presented with user code to enter in browser. DVC will also use this
+  if it cannot launch browser on your behalf.
+
+## Available scopes
```
[the remark this answers is not in the document]
Comment (amritghimire) [reply content/docs/command-reference/studio/login.md:39]: Let’s keep it for proper visibility

File: content/docs/command-reference/studio/index.md [hunk 8]
```diff
@@ -0,0 +1,36 @@
+# studio
+
+Set of commands to authenticate DVC with Studio and save the token to global
+[DVC configuration]: [login](/doc/command-reference/studio/login),
+[logout](/doc/command-reference/studio/logout),
+[token](/doc/command-reference/studio/token),
```
Comment (shcheklein) [inline content/docs/command-reference/studio/index.md:6]: `,` is not needed at the end. Should be `.`.

Set -> A set

File: content/docs/command-reference/studio/index.md [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+
+[dvc configuration]:
+  /doc/user-guide/project-structure/configuration#config-file-locations
+
+## Synopsis
+
+```usage
+usage: dvc studio [-h] [-q | -v] {login,logout,token} ...
```
Comment (shcheklein) [inline content/docs/command-reference/studio/index.md:14]: we need to update syntax highlighter rules for this cc @yathomasi

File: content/docs/command-reference/studio/index.md [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+positional arguments:
+  COMMAND
+    login               Authenticate DVC with Studio host.
+    logout              Logout user from Studio.
+    token               View the token dvc uses to contact Studio.
+```
+
+## Description
+
+`dvc studio` authenticate DVC with Studio and set the token. Once this token has
```
Comment (shcheklein) [inline content/docs/command-reference/studio/index.md:25]: authenticates
sets

File: content/docs/command-reference/studio/index.md [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+been properly configured, DVC will utilize it for seamlessly sharing live
+experiments and sending notifications to Studio regarding any experiments that
```
Comment (shcheklein) [inline content/docs/command-reference/studio/index.md:27]: this token is also about `artifacts get` command for example

File: content/docs/command-reference/studio/login.md [hunk 12]
```diff
@@ -0,0 +1,43 @@
...
+```
+
+## Description
+
+By default, this command authenticates the DVC with Studio using default scopes
+and assigns a random name as the token name.
+
+## Options
+
+- `-h`, `--help` - prints the usage/help message, and exit.
```
Comment (shcheklein) [inline content/docs/command-reference/studio/login.md:21]: those usually go last (I mean -h, -v, etc)

File: content/docs/command-reference/studio/login.md [hunk 13]
```diff
@@ -0,0 +1,43 @@
...
+
+- `-q`, `--quiet` - do not write anything to standard output.
+
+- `-v`, `--verbose` - displays detailed tracing information.
+
+- `-H <hostname>`, `--hostname <hostname>` - The hostname of the Studio instance
```
Comment (shcheklein) [inline content/docs/command-reference/studio/login.md:27]: The -> the 

(and in all other command descriptions)

File: content/docs/command-reference/studio/login.md [hunk 14]
```diff
@@ -0,0 +1,43 @@
...
+  to authenticate with.
+
+- `-s <scopes>`, `--scopes <scopes>` - The scopes for the authentication token.
```
Comment (shcheklein) [inline content/docs/command-reference/studio/login.md:30]: let's put a reference then to the scopes below

File: content/docs/command-reference/studio/token.md [hunk 15]
```diff
@@ -0,0 +1,24 @@
+## studio token
+
+View the token dvc uses to contact Studio
```
Comment (shcheklein) [inline content/docs/command-reference/studio/token.md:3]: DVC

File: content/docs/command-reference/studio/token.md [hunk 16]
```diff
@@ -0,0 +1,24 @@
...
+
+## Synopsis
+
+```usage
+usage: dvc studio token [-h] [-q | -v]
```
Comment (shcheklein) [inline content/docs/command-reference/studio/token.md:8]: what is the use of this command btw? it's not very secure to my mind ..

**Review by shcheklein — COMMENTED** [review]
Looks good, some minor style changes are needed

File: content/docs/command-reference/studio/token.md (hunk 16 above)
Comment (amritghimire) [reply content/docs/command-reference/studio/token.md:8]: It is similar to how we have `gh auth token`. User can use this command in CI or someplace to see the used token and use the value accordingly.

**Review by shcheklein — APPROVED** [review]
@amritghimire LGTM! thanks.

@dberenbaum could you please also do the final check if you have a minute.

**Comment by dberenbaum** [comment]
It looks like deployments are failing due to bad credentials?

**Comment by shcheklein** [comment]
@dberenbaum looks like that was a glitch. It's redeployed now.

File: content/docs/command-reference/studio/index.md [hunk 17]
```diff
@@ -0,0 +1,36 @@
+# studio
+
+A set of commands to authenticate DVC with Studio and save the token to global
```
Comment (dberenbaum) [inline content/docs/command-reference/studio/index.md:3]:
```suggestion
A set of commands to authenticate DVC with [Studio](https://studio.iterative.ai) and save the token to global
```

File: content/docs/command-reference/studio/index.md [hunk 18]
```diff
@@ -0,0 +1,36 @@
...
+    logout              Logout user from Studio.
+    token               View the token DVC uses to contact Studio.
+```
+
+## Description
+
+`dvc studio` authenticates DVC with Studio and sets the token. Once this token
+has been properly configured, DVC will utilize it for seamlessly sharing live
+experiments, sending notifications to Studio regarding any experiments that have
+been pushed and downloading artifacts using `dvc artifacts get`
```
Comment (dberenbaum) [inline content/docs/command-reference/studio/index.md:28]:
```suggestion
been pushed and downloading artifacts using `dvc artifacts get`.
```

File: content/docs/command-reference/studio/login.md [hunk 19]
```diff
@@ -0,0 +1,45 @@
...
+
+## Description
+
+By default, this command authenticates the DVC with Studio using default scopes
+and assigns a random name as the token name.
+
+## Options
+
+- `-H <hostname>`, `--hostname <hostname>` - the hostname of the Studio instance
+  to authenticate with. Defaults to `https://studio.iterative.ai`
```
Comment (dberenbaum) [inline content/docs/command-reference/studio/login.md:22]:
```suggestion
  to authenticate with. Defaults to `https://studio.iterative.ai`.
```

File: content/docs/command-reference/studio/login.md [hunk 20]
```diff
@@ -0,0 +1,45 @@
...
+- `-h`, `--help` - prints the usage/help message, and exit.
+
+- `-q`, `--quiet` - do not write anything to standard output.
+
+- `-v`, `--verbose` - displays detailed tracing information.
+
+## Available scopes
+
+- `EXPERIMENTS` - Experiment operations
+- `DATASETS` - Dataset operations
```
Comment (dberenbaum) [inline content/docs/command-reference/studio/login.md:44]: What is `DATASETS`? Do we need an explanation of scopes in the Studio docs?

**Review by dberenbaum — CHANGES_REQUESTED** [review]
Thanks @amritghimire! I have a few minor edits and one question about scopes.

File: content/docs/command-reference/studio/login.md (hunk 20 above)
Comment (amritghimire) [reply content/docs/command-reference/studio/login.md:44]: Do we have any official documentation for datasets, we can link to? cc. @shcheklein

Comment (amritghimire) [reply content/docs/command-reference/studio/login.md:44]: Yes, it is. I think it is under flag, available for self hosted customer or internal users.

**Comment by dberenbaum** [comment]
What is the overall plan here regarding the `Studio access token` and `Client access tokens`? I worry it's in a confusing state with this PR. We use `Client access tokens` here but don't clarify how they differ from the `Studio access token`, which is referenced a lot throughout the rest of the docs.

File: content/docs/command-reference/studio/login.md (hunk 20 above)
Comment (dberenbaum) [reply content/docs/command-reference/studio/login.md:44]: We can either leave it out of docs for now or leave it in but say something like "coming soon" without going into details.

**Comment by amritghimire** [comment]
> Thoughts on this plan?
> 
> 1. In this PR, add a section [here](https://dvc.org/doc/studio/user-guide/account-and-billing#studio-access-token) to explain the `Client access tokens` and include the scopes there.
> 2. In a follow-up PR, migrate all docs to refer to `Client access tokens` instead of `Studio access token` and deprecate `Studio access token`.

@yathomasi Can you take the the followup?

File: content/docs/command-reference/studio/login.md (hunk 20 above)
Comment (amritghimire) [reply content/docs/command-reference/studio/login.md]: Added the information

**Comment by yathomasi** [comment]
> 2\. migrate all docs to refer to `Client access tokens` instead of `Studio access token` and deprecate `Studio access token`.

We haven't already depricated `Studio access token` in Studio and there are lots of places that uses the `Studio access token` so I think it will take some time and refactoring. So, should we only add `Client access tokens` for now ?

**Comment by dberenbaum** [comment]
> 1. In this PR, add a section [here](https://dvc.org/doc/studio/user-guide/account-and-billing#studio-access-token) to explain the `Client access tokens` and include the scopes there.

Thanks for the update @amritghimire! I don't see this suggestion implemented anywhere. Maybe there's a misunderstanding? I think we should include the client access tokens in the Studio docs in this PR even though we won't drop or change any docs about the old tokens yet.

File: content/docs/command-reference/studio/login.md [hunk 21]
```diff
@@ -40,6 +40,13 @@ and assigns a random name as the token name.
...
 
-- `EXPERIMENTS` - Experiment operations
-- `DATASETS` - Dataset operations
-- `MODELS` - Model registry operations
+- `EXPERIMENTS` - Experiment operations. DVC uses this to share [live
+  experiments] and to notify [Studio](https://studio.iterative.ai/) about
+  [pushed experiments]
+- `MODELS` - Model registry operations like downloading model using
+  `dvc artifacts get`
+- `DATASETS` - Dataset operations. [Coming soon](https://cloud.dvc.ai).
```
Comment (dberenbaum) [inline content/docs/command-reference/studio/login.md:48]: @amritghimire I meant let's move this to https://dvc.org/doc/studio/user-guide/account-and-billing#studio-access-token and add a section explaining these tokens there.

**Review by dberenbaum — APPROVED** [review]

**Pull request merged**